tools/c7n_azure/tests/common.py

```diff
@@ -1,15 +1,170 @@
-import betamax
-import unittest
+import io
+import logging
+import shutil
+import tempfile
+import re
+import six
+from vcr_unittest import VCRTestCase
 
-with betamax.Betamax.configure() as config:
-    config.cassette_library_dir = 'tests/data/cassettes'
+from c7n import policy
+from c7n.schema import generate, validate as schema_validate
+from c7n.ctx import ExecutionContext
+from c7n.utils import CONN_CACHE
+from c7n.resources import load_resources
+from c7n_azure.session import Session
 
-    
-class BaseTest(unittest.TestCase):
+load_resources()
+C7N_SCHEMA = generate()
+DEFAULT_SUBSCRIPTION_ID = 'ea42f556-5106-4743-99b0-c129bfa71a47'
+
+
+class AzureVCRBaseTest(VCRTestCase):
+
+    def _get_vcr_kwargs(self):
+        return super(VCRTestCase, self)._get_vcr_kwargs(
+            filter_headers=['Authorization',
+                            'client-request-id',
+                            'retry-after',
+                            'x-ms-client-request-id',
+                            'x-ms-correlation-request-id',
+                            'x-ms-ratelimit-remaining-subscription-reads',
+                            'x-ms-request-id',
+                            'x-ms-routing-request-id',
+                            'x-ms-gateway-service-instanceid',
+                            'x-ms-ratelimit-remaining-tenant-reads',
+                            'x-ms-served-by', ],
+            before_record_request=self.request_callback
+        )
+
+    def _get_vcr(self, **kwargs):
+        myvcr = super(VCRTestCase, self)._get_vcr(**kwargs)
+        myvcr.register_matcher('azurematcher', self.azure_matcher)
```

Comment:
is there a compatible option on the vcr serialization to use json?, yaml is significantly more expensive to parse (especially if the install is missing the c extension)? otoh it looks there are some direct binary strings in the yaml serialization so its unclear if thats compatible or not.

Reply:
Yep it'll do JSON as well.  We can try that.